src/Properties/ModelAccessorExtension.php

```diff
@@ -21,13 +25,50 @@ public function hasProperty(ClassReflection $classReflection, string $propertyNa
             return false;
         }
 
+        $camelCase = Str::camel($propertyName);
+
+        if ($classReflection->hasNativeMethod($camelCase)) {
+            $methodReflection = $classReflection->getNativeMethod($camelCase);
+
+            if ($methodReflection->isPublic() || $methodReflection->isPrivate()) {
+                return false;
+            }
+
+            $returnType = ParametersAcceptorSelector::selectSingle($methodReflection->getVariants())->getReturnType();
+
+            if (! $returnType instanceof GenericObjectType) {
+                return false;
+            }
```

Comment:
@canvural, thanks for working on this! However, this is causing all of my L9 accessors (i.e., attributes that are not mutating a table column) to fail with an 'undefined property' error and it took me nearly a whole day to figure out why---I didn't know it needed the `@return Attribute` generic.

```php
    /** @return Attribute<string, never> */        <----- UNDOCUMENTED
    protected function someVar(): Attribute
    {
        return Attribute::make(
            get: fn (): string => 'fooBar',
        );
    }
```

Also, I'm still trying to get PHPStan `Level 1` to pass which is checking the following:

> Level 1: possibly undefined variables, unknown magic methods and properties on classes with __call and __get

but fails on attributes missing the generic which seems unnecessarily strict as `Level 1` is just concerned about the property *existence* and not the type. I would expect this to fail around `Level 6: report missing typehints`.

Would it be possible to only enforce the generic check at a higher level?

Thanks!

Reply:
@CalebDW I agree we don't have enough documentation of the features here. This Attribute change was only in release notes I guess.

> Also, I'm still trying to get PHPStan Level 1 to pass which is checking the following:
	Level 1: possibly undefined variables, unknown magic methods and properties on classes with __call and __get
but fails on attributes missing the generic which seems unnecessarily strict as Level 1 is just concerned about the property existence and not the type. I would expect this to fail around Level 6: report missing typehints.
	Would it be possible to only enforce the generic check at a higher level?

Larastan does not have any influence on error levels. Those come from PHPStan. Which error exactly you are referring to?

Reply:
@canvural,

The error I've been getting for Attributes without the `@return Attribute<>` generic is:

```
Access to an undefined property Path\To\Model::$foo_bar.
```

But I don't think this makes sense because Larastan can detect that the attribute exists without the generic PHPdoc right? You just wouldn't have any type information about it.

Reply:
The generic return type is needed. That's how Larastan recognizes that this method is an attribute.

Reply:
But `ModelPropertyExtension::hasAttribute` uses the following check to determine if a property is an Attribute (as does this same method).

```php
            if (! (new ObjectType(Attribute::class))->isSuperTypeOf($returnType)->yes()) {
                return false;
            }
```

Is this not enough to determine existence?